FPGA: Remove the dependency on the `dpc_common.hpp` file

yuguen-intel opened the pull request:
# Existing Sample Changes
## Description

This changes removes the dependency on the `dpc_common.hpp` file for all FPGA code samples.
This file contained two things:
- an exception handler: it has been reimplemented in the include directory of the FPGA part of the samples
- a timer: the use of this timer has been replaced with regular `std::chrono` function calls


## How Has This Been Tested?

- [x] Command Line

jkinsky reviewed (approved):
All the changes to the readme.md files are consistent with the overall removal of the header dependence. Looks good.

yuguen-intel commented:
This change made all the fpga code samples to rely on the include directory.
This required changes to all the sample.son files for the code samples browsers to fetch the include directory as well as the individual samples.
This also addresses some issues found in the 2022.3 release that the anr and db design don't fetch the include directory
